File: storage-snippets/copy_file_from_workspace_bucket.R
# This snippet assumes that you run setup first

# This code copies a file from your Google Bucket into a dataframe

# replace 'test.csv' with the name of the file in your google bucket (don't delete the quotation marks)
name_of_file_in_bucket <- 'test.csv'

########################################################################
##
################# DON'T CHANGE FROM HERE ###############################
##
########################################################################

# Get the bucket name
my_bucket <- Sys.getenv('WORKSPACE_BUCKET')

# Copy the file from current workspace to the bucket
system(paste0("gsutil cp ", my_bucket, "/notebooks/data/", name_of_file_in_bucket, " ."), intern=T)

# Load the file into a dataframe
my_dataframe  <- read_csv(name_of_file_in_bucket)
head(my_dataframe)

File: storage-snippets/copy_file_from_workspace_bucket.py
# This snippet assumes you run setup first

# This code copies file in your Google Bucket and loads it into a dataframe

# Replace 'test.csv' with THE NAME of the file you're going to download from the bucket (don't delete the quotation marks)
name_of_file_in_bucket = 'test.csv'

########################################################################
##
################# DON'T CHANGE FROM HERE ###############################
##
########################################################################

# get the bucket name
my_bucket = os.getenv('WORKSPACE_BUCKET')

# copy csv file from the bucket to the current working space
args = ["gsutil", "cp", f"{my_bucket}/data/{name_of_file_in_bucket}", "."]
output = subprocess.run(args, capture_output=True)

if output.returncode != 0:
    # failed to copy the file - add appropriate error handling here
    print(output.stderr)
else:
    print(f'[INFO] {name_of_file_in_bucket} is successfully downloaded into your working space')
    my_dataframe = pd.read_csv(name_of_file_in_bucket)
    my_dataframe.head()
